share/pvss/pvss.go

```diff
@@ -35,52 +35,65 @@ var errorTooFewShares = errors.New("not enough shares to recover secret")
 var errorDifferentLengths = errors.New("inputs of different lengths")
 var errorEncVerification = errors.New("verification of encrypted share failed")
 var errorDecVerification = errors.New("verification of decrypted share failed")
+var errorChallengeVerification = errors.New("verification of the challenge failed")
```

Comment:
I'd suggest exposing these errors and prefixing them with just `Err` as per the [go naming conventions](https://littleroot.org/blog/go-naming-errors/), e.g. `ErrChallengeVerification`.
That will allow users to react to errors in a meaningful way without doing string matching